tensorflow_federated/python/learning/keras_utils.py

```diff
@@ -111,14 +123,49 @@ def from_keras_model(keras_model,
     ValueError: If `keras_model` was compiled.
   """
   py_typecheck.check_type(keras_model, tf.keras.Model)
+  py_typecheck.check_type(loss, (tf.keras.losses.Loss,
+                                 collections.Sequence,
+                                 collections.Mapping))
+
+  if loss_weights is not None:
+    py_typecheck.check_type(loss, (collections.Sequence,
+                                   collections.Mapping))
+
+  if isinstance(loss, collections.Sequence):
+    if len(loss) != len(keras_model.outputs):
+      raise ValueError('`keras_model` must have equal number of '
+                       'outputs and losses')
+
+    if loss_weights is not None and len(loss) != len(loss_weights):
+      raise ValueError('`keras_model` must have equal number of '
+                       'losses and loss weights')
+
   if keras_model._is_compiled:  # pylint: disable=protected-access
     raise ValueError('`keras_model` must not be compiled. Use '
                      'from_compiled_keras_model() instead.')
+
   dummy_tensors = _preprocess_dummy_batch(dummy_batch)
   if optimizer is None:
+    if isinstance(loss, collections.Mapping):
+      loss_functions = []
+      for name in keras_model.output_names:
+        if name not in loss:
+          logging.warning(
+              'Output {0} missing from loss dictionary. We assume '
+              'this was done on purpose. The fit and evaluate APIs will not be '
+              'expecting any data to be passed to {0}.'.format(name))
+        loss_functions.append(loss[name])
+
+    elif isinstance(loss, collections.Sequence):
+      loss_functions = loss
+
```

Comment:
Is `yapf` adding the newlines before each new condition branch? This isn't common in TFF, if possible lets remove the empty newlines.